### PR TITLE
feat: add label style to segmented buttons

### PR DESCRIPTION
**Pull request by rKatex**
### Summary

Currently there is no way to change the style of the label of a `SegmentedButton`. This PR introduces the ability to give each element of the `buttons`-list an individual style.

```tsx
{
    value: 'test',
    label: 'Test',
    style: {
        // Some wrapper style
    },
    labelStyle: {
        fontSize: 5,
    },
}
```

### Test plan

Like in the example above, just apply some label style to a specific `SegementedButton` item.

### CODE REVIEW

File: src/components/SegmentedButtons/SegmentedButtonItem.tsx [hunk 1]
```diff
@@ -80,6 +80,7 @@ export type Props = {
    */
   density?: 'regular' | 'small' | 'medium' | 'high';
   style?: StyleProp<ViewStyle>;
+  labelStyle?: StyleProp<TextStyle>;
```
Comment (lukewalczak) [inline src/components/SegmentedButtons/SegmentedButtonItem.tsx:83]: Please add a prop description

Comment (rKatex) [reply src/components/SegmentedButtons/SegmentedButtonItem.tsx:86]: Is the description I added OK?

File: src/components/__tests__/SegmentedButton.test.tsx [hunk 2]
```diff
@@ -35,6 +35,21 @@ it('renders disabled segmented button', () => {
   expect(tree).toMatchSnapshot();
 });
 
+it('applies labelStyle for each segmented button', () => {
```
[the remark this answers is not in the document]
Comment (rKatex) [reply src/components/__tests__/SegmentedButton.test.tsx]: I added two new tests in [SegmentedButton.test.tsx:443](https://github.com/callstack/react-native-paper/pull/3926/files#diff-343124f377aa2894adca3a4273e36fd02f9bf405ff93a65f469120db0d07f1af) and removed the snapshot test. I hope thats what you meant.

File: src/components/SegmentedButtons/SegmentedButtonItem.tsx [hunk 3]
```diff
@@ -80,6 +80,10 @@ export type Props = {
    */
   density?: 'regular' | 'small' | 'medium' | 'high';
   style?: StyleProp<ViewStyle>;
+  /**
+   * Allows you to customize the style of the label associated with a SegmentedButton.
```
Comment (lukewalczak) [inline src/components/SegmentedButtons/SegmentedButtonItem.tsx:84]:
```suggestion
   * Style for the button label.
```

File: src/components/SegmentedButtons/SegmentedButtonItem.tsx [hunk 4]
```diff
@@ -216,9 +221,10 @@ const SegmentedButtonItem = ({
           ) : null}
           <Text
             variant="labelLarge"
-            style={[styles.label, labelTextStyle]}
+            style={[styles.label, labelTextStyle, labelStyle]}
             selectable={false}
             numberOfLines={1}
+            testID={testID ? `${testID}-label` : undefined}
```
Comment (lukewalczak) [inline src/components/SegmentedButtons/SegmentedButtonItem.tsx:227]:
```suggestion
            testID={`${testID}-label`}
```

**Review by lukewalczak — APPROVED** [review]

**Pull request merged**